funsor/distribution.py

```diff
@@ -129,6 +143,23 @@ def has_rsample(self):
     def has_enumerate_support(self):
         return getattr(self.dist_class, "has_enumerate_support", False)
 
+    @classmethod
+    def eager_log_prob(cls, *params):
+        params, value = params[:-1], params[-1]
+        params = params + (Variable("value", value.output),)
+        instance = reflect(cls, *params)
+        raw_dist, value_name, value_output, dim_to_name = instance._get_raw_dist()
```

Comment:
I had to refactor `eager_log_prob` to use `Distribution._get_raw_dist()` to get the new tests to pass.